crates/daphne-server/src/lib.rs

```diff
@@ -100,6 +112,59 @@ impl router::DaphneService for App {
     fn signing_key(&self) -> Option<&p256::ecdsa::SigningKey> {
         self.service_config.signing_key.as_ref()
     }
+
+    async fn check_bearer_token(
+        &self,
+        presented_token: &BearerToken,
+        sender: DapSender,
+        task_id: TaskId,
+        is_taskprov: bool,
```

Comment:
My goal here was to avoid passing in the entire DapRequest just to check some of it's fields. It's specially awkward since the `sender` may not be present in the DapRequest and then this method has to deal with that. 

I don't like this bool parameter, but it felt like the least bad option